File: only-buns-be/src/main/java/com/example/onlybunsbe/model/ChatMessage.java
package com.example.onlybunsbe.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "chat_messages")
public class ChatMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender; // Korisnik koji šalje poruku

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id", nullable = false)
    private GroupChat group; // Grupni čet

    @NotNull
    @Column(name = "content", nullable = false, length = Integer.MAX_VALUE)
    private String content;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private MessageType type;

    @NotNull
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt = Instant.now();

    public enum MessageType {
        CHAT, JOIN, LEAVE
    }
}

File: only-buns-fe/src/app/shared/model/chatMessage.ts
export interface ChatMessage {
  id?: number;           // ID poruke
  content: string;      // Sadržaj poruke
  createdAt: string;         // Vreme slanja poruke
  senderId: number;     // ID pošiljaoca
  groupId?: number;     // ID grupe (null za privatne poruke)
  type: MessageType;    // Tip poruke (CHAT, JOIN, LEAVE)
}

export type MessageType = 'CHAT' | 'JOIN' | 'LEAVE';
